api/opencl_runtime_layer.asciidoc

```diff
@@ -10911,112 +10911,90 @@ include::{generated}/api/protos/clSetKernelArg.txt[]
 include::{generated}/api/version-notes/clSetKernelArg.asciidoc[]
 
   * _kernel_ is a valid kernel object.
-  * _arg_index_ is the argument index.
-    Arguments to the kernel are referred by indices that go from 0 for the
-    leftmost argument to _n_ - 1, where _n_ is the total number of arguments
-    declared by a kernel (see below).
-  * _arg_size_ specifies the size of the argument value.
-    If the argument is a memory object, the _arg_size_ value must be equal to
-    `sizeof({cl_mem_TYPE})`.
-    For arguments declared with the `local` qualifier, the size specified will
-    be the size in bytes of the buffer that must be allocated for the `local`
-    argument.
-    If the argument is of type _sampler_t_, the _arg_size_ value must be equal
-    to `sizeof({cl_sampler_TYPE})`.
-    If the argument is of type _queue_t_, the _arg_size_ value must be equal to
-    `sizeof({cl_command_queue_TYPE})`.
-    For all other arguments, the size will be the size of argument type.
-  * _arg_value_ is a pointer to data that should be used as the argument value
-    for argument specified by _arg_index_.
+  * _arg_index_ is the kernel argument index.
+    Kernel arguments are referred to by indices that go from zero to
+    _n - 1_, where _n_ is the total number of arguments declared by the kernel.
+  * _arg_size_ specifies the size of the kernel argument value.
+  * _arg_value_ is a pointer to the data for the kernel argument.
     The argument data pointed to by _arg_value_ is copied and the _arg_value_
     pointer can therefore be reused by the application after {clSetKernelArg}
     returns.
-    The argument value specified is the value used by all API calls that enqueue
-    _kernel_ ({clEnqueueNDRangeKernel} and {clEnqueueTask}) until the argument
-    value is changed by a call to {clSetKernelArg} for _kernel_.
+    The argument data is used by all API calls that enqueue the kernel until the
+    argument is changed by another call to {clSetKernelArg} for the kernel.
+
+If the kernel argument being set is a pointer to the `global` or `constant`
+address space, then _arg_value_ must point to a buffer memory object or `NULL`,
+or _arg_value_ must be `NULL`.
+If _arg_value_ is `NULL` or points to `NULL`, then the kernel argument will be
+set to `NULL`.
+
+If the kernel argument being set is a pointer to the `constant` address space,
+then the size in bytes of the memory object cannot exceed
+{CL_DEVICE_MAX_CONSTANT_BUFFER_SIZE}.
+
+// TODO: Should CL_DEVICE_MAX_CONSTANT_BUFFER_SIZE be checked when the kernel
+// argument is set or when the kernel is enqueued?
+// If it is checked when the kernel argument is set, then what is the error code
+// if the size is exceeded?
```

Comment:
Note that this requirement was documented previously in text, but it was not described as any specific error condition.

If we check for this error when the kernel argument is set, what should the expected behavior be for a multi-device context, where the size is exceeded for one of the devices, but not for the other?

Even though it would mean detecting the error later, we may want to check for this error when the kernel is enqueued rather than when the kernel argument is set.